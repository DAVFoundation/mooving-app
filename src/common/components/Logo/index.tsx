import React from 'react';
import { Component } from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { customSizes } from '../../styles';

const styles = StyleSheet.create({
  logoStandard: {
    width: customSizes.main * 1.5,
    height: customSizes.main * 1.5,
  },
});

interface ILogoProps {
  isDark: boolean;
}

export default class Logo extends Component<ILogoProps> {
  public render() {

    const logo = this.props.isDark
      ? require('../../assets/logos/logo-dav_dark.png')
      : require('../../assets/logos/logo-dav_light.png');

    return (
      <View>
        <Image style={styles.logoStandard} source={logo} />
      </View>
    );
  }
}
