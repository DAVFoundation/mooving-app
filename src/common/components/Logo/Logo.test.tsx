import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../Logo';

it('renders correctly with defaults', () => {
  const view = renderer
    .create(<Logo isDark/>)
    .toJSON();
  expect(view).toMatchSnapshot();
});
