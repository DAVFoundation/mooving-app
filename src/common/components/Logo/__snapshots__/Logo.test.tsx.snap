// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/common/assets/logos/logo-dav_dark.png",
      }
    }
    style={
      Object {
        "height": 82.5,
        "width": 82.5,
      }
    }
  />
</View>
`;
